scheduler/task_priority_cmp.go

```diff
@@ -133,13 +133,12 @@ type byTaskGroupOrder struct{}
 func (c *byTaskGroupOrder) name() string { return "order within task group" }
 func (c *byTaskGroupOrder) compare(t1, t2 task.Task, _ *CmpBasedTaskComparator) (int, string, error) {
 	// Try other comparators if both tasks are not in task groups
-	reason := "neither task in a group"
 	if t1.TaskGroup == "" && t2.TaskGroup == "" {
 		return 0, "", nil
 	}
 
 	// If one task is in a task group, sort that one higher, which keeps the pre-byTaskGroupOrder order.
-	reason = "higher task is in a group"
+	reason := "one is in a group and one is not"
```

Comment:
"one is in a group and one is not, so one in group is first"

Reply:
changed to "the task in a task group is first"